Firestore/core/src/firebase/firestore/util/status_win.cc

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2018 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "Firestore/core/src/firebase/firestore/util/status.h"
+
+#if defined(_WIN32)
+
+#include "Firestore/core/src/firebase/firestore/util/string_format.h"
+#include "Firestore/core/src/firebase/firestore/util/string_win.h"
+
+namespace firebase {
+namespace firestore {
+namespace util {
+
+/**
+ * Returns the Canonical error code for the given Windows API error code as
+ * obtained from GetLastError().
+ */
+static FirestoreErrorCode CodeForLastError(DWORD error) {
+  switch (error) {
+    case ERROR_SUCCESS:
+      return FirestoreErrorCode::Ok;
+
+    case ERROR_INVALID_ACCESS:
+      return FirestoreErrorCode::Internal;
```

Comment:
This sounds like it could be PermissionDenied?

https://docs.microsoft.com/en-us/windows/desktop/debug/system-error-codes--0-499- is supremely unhelpful.

"ERROR_INVALID_ACCESS

12 (0xC)

The access code is invalid."

I assume that's referring to some sort of auth token. Though maybe the reasoning here is that it isn't *our* auth token, so something else must have gone horribly wrong?

Reply:
Good call. I had confused this with something else.